scripts/common/proxy.sh

```diff
@@ -47,6 +47,46 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# the following code will check if the proxy is invalid by doing a check as following
+# ~$ sudo HTTP_PROXY=http://77.89.35.50:8080 HTTPS_PROXY=http://77.89.35.50:8080 ctr image pull docker.io/library/nginx:latest
+# docker.io/library/nginx:latest: resolving      |--------------------------------------|
+# elapsed: 29.9s                  total:   0.0 B (0.0 B/s)
+# INFO[0030] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/nginx/manifests/latest\": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout" host=registry-1.docker.io
+# ctr: failed to resolve reference "docker.io/library/nginx:latest": failed to do request: Head "https://registry-1.docker.io/v2/library/nginx/manifests/latest": proxyconnect tcp: dial tcp 77.89.35.50:8080: i/o timeout
+function check_proxy_config() {
+    if [ -n "$CONTAINERD_VERSION" ]; then
+        logStep "Checking proxy configuration with Containerd"
+
+        # Echo containerd Proxy config:
+        local proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+        if [ -f "$proxy_config_file" ]; then
+            echo ""
+            log "Proxy config:"
+            grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+            echo ""
+        else
+            log "Skipping test. No HTTP proxy configuration found."
+            return
+        fi
+
+        # Get proxy configuration from http-proxy.conf file
+        local http_proxy=$(grep -E '^\s*Environment\s*=\s*"HTTP_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"HTTP_PROXY=(.*)"/\1/')
+        local https_proxy=$(grep -E '^\s*Environment\s*=\s*"HTTPS_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"HTTPS_PROXY=(.*)"/\1/')
+        local no_proxy=$(grep -E '^\s*Environment\s*=\s*"NO_PROXY=' "$proxy_config_file" | sed -E 's/^\s*Environment\s*=\s*"NO_PROXY=(.*)"/\1/')
+
+        # Make the Docker image pull call with proxy configuration
+        if response=$(sudo ctr image pull --http-proxy "$http_proxy" --https-proxy "$https_proxy" --no-proxy "$no_proxy" docker.io/library/nginx:latest 2>&1) && [[ $response == *"proxy"* ]]; then
```

Comment:
I think we could use an image we own for this, or at least a smaller one. What do you think ?

Reply:
The image actually does not matter, so I update it to make clear. We check the error and we just inform that when the problem is proxy. If the error is image does not exist and etc then it will not raise the warning.

See:

```
⚙  Checking proxy configuration with Containerd

Proxy config:
Environment="HTTP_PROXY=http://invalidproxy:1234"
Environment="HTTPS_PROXY=http://invalidproxy:1234"

Proxy connection issues were identified
Please review the proxy configuration and ensure that it is valid.
More info: https://kurl.sh/docs/install-with-kurl/proxy-installs
```

The error for this call is:

```
~$ sudo HTTP_PROXY=http://invalidproxy:1234 HTTPS_PROXY=http://invalidproxy:1234 NO_PROXY= ctr image pull test/invalid/image:latest
INFO[0000] trying next host                              error="failed to do request: Head \"https://test/v2/invalid/image/manifests/latest\": proxyconnect tcp: dial tcp: lookup invalidproxy: Temporary failure in name resolution" host=test
ctr: failed to resolve reference "test/invalid/image:latest": failed to do request: Head "https://test/v2/invalid/image/manifests/latest": proxyconnect tcp: dial tcp: lookup invalidproxy: Temporary failure in name resolution
```

Reply:
But that message is helpful "Temporary failure in name resolution"

Other messages could be "Connection timeout", etc...

Reply:
I see. Your suggestion is addressed now, see:

```sh
⚙  Checking proxy configuration with Containerd

Proxy config:
Environment="HTTP_PROXY=http://invalidproxy:1234"
Environment="HTTPS_PROXY=http://invalidproxy:1234"

Proxy connection issues were identified:
Proxy error: Temporary failure in name resolution

Please review the proxy configuration and ensure that it is valid.
More info: https://kurl.sh/docs/install-with-kurl/proxy-installs
```